packages/core/src/base-chart.ts

```diff
@@ -954,4 +1000,30 @@ export class BaseChart {
 	getBBox(selector: any) {
 		return this.innerWrap.select(selector).node().getBBox();
 	}
+// ================================================================================
+	// Makes chart data components more screen reader friendly
+	// ================================================================================
+	makeAccessible (element?: any, dataList?: any) {
+		// Make chart data components tabbable
+		element.attr("tabindex", 0);
+		// Aria label contains the x value, y value, and dataset for the data element
+		if (element.attr("class") === "bar" || element.attr("class") === "dot") {
+			element.attr("aria-label", (d) => `Label: ${d.label}, Value: ${d.value}, belongs to ${d.datasetLabel}`);
+		// Slices
+		} else if (dataList !== undefined) {
+			element.attr("aria-label", (d) => `Label: ${d.data.label}, Value: ${d.value},
+			percentage is ${Tools.convertValueToPercentage(d.data.value, dataList)}`);
+		// Legend items
+		} else if (element.attr("class") === "legend-circle") {
+			element.attr("aria-label", (d) => `Legend Item: ${d.key}, Status: ${d.value ? "enabled" : "disabled"},
+			Click to ${d.value ? "disable" : "enable"}`);
+		// Chart containers
+		} else if (element.attr("chart-id")) {
+			element.attr("aria-label", `${element.attr("chart-id")}`);
+		// Legend containers
+		} else if (element.attr("class") === "legend-wrapper") {
+			element.attr("aria-label", "Legend - click items to add or remove them from the chart");
```

Comment:
We need a way to make these hard-coded labels translatable into other languages.
In carbon-components-angular, we do that (to simplify) by providing inputs with defaults read from json files. Here, those could be config/options params.

Probably not a do-or-die for this PR. Definitely something we need to address. cc @theiliad @cal-smith